lib/portage/_emirrordist/FetchTask.py

```diff
@@ -551,6 +552,23 @@ def _fetch_copier_exit(self, copier):
 			self.returncode = 1
 			self.wait()
 
+	def _make_layout_symlinks(self):
+		dist_path = None
+		for layout in self.config.layouts[1:]:
+			if dist_path is None:
+				dist_path = os.path.join(self.config.options.distfiles,
+						self.config.layouts[0].get_path(self.distfile))
+			sym_path = os.path.join(self.config.options.distfiles,
+					layout.get_path(self.distfile))
+			ensure_dirs(os.path.dirname(sym_path))
+			# TODO: how to handle errors?
+			os.symlink(os.path.relpath(dist_path, os.path.dirname(sym_path)),
+					sym_path)
```

Comment:
Using symlinks rather than hardlinks so that mirrors can avoid the overhead of rsync --hard-links? When eliminating the flat structure, mirrors will have to re-transfer all files at that point?

Reply:
More like because we don't know what mirror admins are using. `-l` is part of `-a` (but `-H` is not), so I think there's a better chance of it actually working.

And yes, I presume that eliminating that structure will have major impact. That's why my plan involves eliminating it a few directories at a time, to reduce the change that users will have major problems using the mirrors while they update.

Of course the final decision is up to @gentoo/infra. I wonder if we should have a command-line option to decide between symlinks and hardlinks.